Tests/BootCamp.Chapter.Tests/DeckTest.cs

```diff
@@ -0,0 +1,50 @@
+﻿using BootCamp.Chapter.Gambling;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace BootCamp.Chapter.Tests
+{
+    public class DeckTest
+    {
+        // I cannot test the happy path on the constructor because the variable that  holds the contents is private
+
+        [Fact]
+        public void New_Given_Null_List_Throws_NullArgumentException()
+        {
+            Assert.Throws<ArgumentNullException>(() => new Deck(null));
+        }
+
+        [Fact]
+        public void DrawFromTop_Given_Empty_List_Throws_OutOfCardException()
+        {
+            var deck = new Deck(new List<Card>() { });
+
+            Assert.Throws<OutOfCardsException>(() => deck.DrawFromTop());
+        }
+
+        [Theory]
+        [MemberData(nameof(deckExpectations))]
+        public void DrawFromTop_Given_List_Return_Last_Item(Deck deck, Card expected)
+        {
+            var actual = deck.DrawFromTop();
+
+            Assert.Equal(expected, actual);
+        }
+
+        // DrawRandom cannot be tested because it needed a mock and we did not learn that on lesson1.
+
+        // Shuffle cannot be tested because there is not way I can make a test that always returns the same.
+
+        // ValidateDeckSize cannot be tested because the method is private.
```

Comment:
Which is fine, because you already tested it by testing draw methods.

Reply:
Not every function has to be tested directly.

Reply:
oke, learned another thing